woningwaardering/stelsels/zelfstandige_woonruimten/keuken/keuken_2024.py

```diff
@@ -0,0 +1,158 @@
+            WoningwaarderingResultatenWoningwaarderingResultaat | None
+        ) = None,
+    ) -> WoningwaarderingResultatenWoningwaarderingGroep:
+        woningwaardering_groep = WoningwaarderingResultatenWoningwaarderingGroep(
+            criteriumGroep=WoningwaarderingResultatenWoningwaarderingCriteriumGroep(
+                stelsel=Woningwaarderingstelsel.zelfstandige_woonruimten.value,
+                stelselgroep=Woningwaarderingstelselgroep.keuken.value,
+            )
+        )
+        woningwaardering_groep.woningwaarderingen = []
+
+        geldige_ruimtes = [
+            ruimte
+            for ruimte in eenheid.ruimten or []
+            # check of een ruimte behoort tot te ruimtesoorten die in aanmerking komen voor de stelselgroep keuken
+            if ruimte.detail_soort
+            and (
+                ruimte.detail_soort.code == Ruimtedetailsoort.keuken.code
+                or ruimte.detail_soort.code
+                == Ruimtedetailsoort.woonkamer_en_of_keuken.code
+                or ruimte.detail_soort.code == Ruimtedetailsoort.woonkamer.code
+                or ruimte.detail_soort.code
+                == Ruimtedetailsoort.woon_en_of_slaapkamer.code
+                or ruimte.detail_soort.code == Ruimtedetailsoort.slaapkamer.code
+            )
+        ]
+
+        if not geldige_ruimtes:
+            logger.warning(
+                f"Kan geen punten geven voor stelselgroep {Woningwaarderingstelselgroep.keuken.naam}: Geen geldige ruimte detailsoort gevonden in eenheid {eenheid.id}."
+                f"Geldige ruimtedetailsoorten: {Ruimtedetailsoort.keuken.naam}, {Ruimtedetailsoort.woonkamer_en_of_keuken.naam}, {Ruimtedetailsoort.woonkamer.naam}, {Ruimtedetailsoort.woon_en_of_slaapkamer.naam} {Ruimtedetailsoort.slaapkamer.naam}."
+            )
+            return woningwaardering_groep
+
+        ruimten_met_aanrecht = [
+            ruimte
+            for ruimte in geldige_ruimtes
+            # check of de ruimte een aanrecht heeft
+            if ruimte.bouwkundige_elementen
+            for bouwkundig_element in ruimte.bouwkundige_elementen
+            if bouwkundig_element.detail_soort
+            and bouwkundig_element.detail_soort.code
+            == Bouwkundigelementdetailsoort.aanrecht.code
+        ]
+
+        if not ruimten_met_aanrecht:
+            logger.warning(
+                f"Kan geen punten geven voor stelselgroep {Woningwaarderingstelselgroep.keuken.naam}: Geen ruimte met een {Bouwkundigelementdetailsoort.aanrecht.naam} gevonden voor eenheid {eenheid.id}"
+            )
+            return woningwaardering_groep
+
+        if ruimten_met_aanrecht:
+            ruimten_met_aanrecht_lengte = [
+                ruimte
+                for ruimte in ruimten_met_aanrecht
+                if ruimte.bouwkundige_elementen
+                # Check of de aanrechten een lengte hebben
+                for aanrecht in ruimte.bouwkundige_elementen
+                if aanrecht.lengte
+            ]
```

Comment:
Hier zitten alle bouwkundige elementen nog in, niet alleen de aanrechten. Daar moeten we hier dus alsnog op filteren. Met de Tuple van ruimte en aanrecht waar de ruimten_met_aanrecht definieren, hoeven we er maar 1x doorheen en heb je nog steeds de relatie tussen ruimte en aanrecht die we nodig hebben voor de log messages.
```suggestion
        if ruimten_met_aanrecht:
            ruimten_met_aanrecht_lengte = [
                (ruimte, aanrecht)
                for (ruimte, aanrecht) in ruimten_met_aanrecht
                if aanrecht.lengte
            ]
```